### PR TITLE
Storing questions and answers in db and their relationship is establi…

### PR DESCRIPTION
**Pull request by muhammad-naveed-brainx**
…shed

### CODE REVIEW

File: app/Http/Controllers/AnswerController.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Answer;
+use Illuminate\Http\Request;
+
+class AnswerController extends Controller
+{
+    public function addAns($id)
```
Comment (faizan-khalid-brainx) [inline app/Http/Controllers/AnswerController.php]: addAnswer is not a restful name it should be named store

File: app/Http/Controllers/AnswerController.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        $ans = new Answer();
+        $ans->body = \request('ans-body');
```
Comment (faizan-khalid-brainx) [inline app/Http/Controllers/AnswerController.php]: shouldn't use kebab case, either use camel case or snake case

Comment (faizan-khalid-brainx) [inline app/Http/Controllers/AnswerController.php]: should use complete words

File: app/Http/Controllers/AnswerController.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+namespace App\Http\Controllers;
+
+use App\Models\Answer;
+use Illuminate\Http\Request;
+
+class AnswerController extends Controller
+{
+    public function addAns($id)
+    {
+        $ans = new Answer();
```
Comment (faizan-khalid-brainx) [inline app/Http/Controllers/AnswerController.php]: should use complete words

File: app/Http/Controllers/QuestionController.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Question;
+
+class QuestionController extends Controller
+{
+    public function add()
```
Comment (faizan-khalid-brainx) [inline app/Http/Controllers/QuestionController.php]: names should be restful should be store

File: app/Http/Controllers/QuestionController.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        $q = new Question();
+        $q->title= \request('q-title');
+        $q->body = \request('q-body');
+        $q->save();
+
+        return redirect()->action([QuestionController::class, 'showAll']);
+    }
+
+    public function showAll()
```
Comment (faizan-khalid-brainx) [inline app/Http/Controllers/QuestionController.php]: first use all restful name index is available

File: app/Http/Controllers/QuestionController.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Question;
+
+class QuestionController extends Controller
+{
+    public function add()
+    {
+        $q = new Question();
```
Comment (faizan-khalid-brainx) [inline app/Http/Controllers/QuestionController.php]: q is not a valid variable name

File: app/Http/Controllers/QuestionController.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        $q->title= \request('q-title');
+        $q->body = \request('q-body');
```
Comment (faizan-khalid-brainx) [inline app/Http/Controllers/QuestionController.php]: q-title is kebab case and not a valid name

File: public/css/app.css [hunk 8]
```diff
@@ -9,4 +9,14 @@ p{
     line-height: 1.6;
     text-align: justify;
 }
+a{
+    text-decoration: none;
+}
+
+/* this means that all articles except the first one*/
```
Comment (faizan-khalid-brainx) [inline public/css/app.css]: misleading comment. this css is applies to all articles immediately after article

File: resources/views/ask_question.blade.php [hunk 9]
```diff
@@ -4,7 +4,8 @@
     <div class="container mt-3">
         <div class="p-3"><h2>Ask a public question</h2></div>
 
-        <form action="/">
+        <form action="/ask-question" method="post">
```
Comment (faizan-khalid-brainx) [inline resources/views/ask_question.blade.php:7]: use named route here it encourage decoupling

File: resources/views/question.blade.php [hunk 10]
```diff
@@ -2,10 +2,45 @@
...
+                <div class="mb-0"><h1><i class="bi bi-caret-up-fill"></i></h1></div>
+                <div><h1>{{$question->vote}}</h1></div>
+                <div><h1><i class="bi bi-caret-down-fill"></i></h1></div>
+            </div>
+            <div class="col-sm-11"><p> {!! $question->body !!} </p></div>
         </div>
     </article>
+{{--    answers--}}
+    <div class="d-flex flex-column">
+        <div><h4>{{$question->count()}} Answers</h4></div>
```
Comment (faizan-khalid-brainx) [inline resources/views/question.blade.php]: lazy loading count on frontend. Violating MVC and unnecessary performance overhead

File: resources/views/question.blade.php [hunk 11]
```diff
@@ -2,10 +2,45 @@
...
+        @foreach($question->answers as $ans)
```
Comment (faizan-khalid-brainx) [inline resources/views/question.blade.php]: lazy loading all the answers in loop and violating MVC

File: resources/views/question.blade.php [hunk 12]
```diff
@@ -2,10 +2,45 @@
...
+        <article>
+            <div class="row">
+                <div class="col-sm-1 d-flex flex-column justify-content-center">
+                    <div class="mb-0"><h1><i class="bi bi-caret-up-fill"></i></h1></div>
+                    <div><h1>{{$ans->vote}}</h1></div>
```
Comment (faizan-khalid-brainx) [inline resources/views/question.blade.php]: lazy loading vote in loop

File: resources/views/question.blade.php [hunk 13]
```diff
@@ -2,10 +2,45 @@
...
+            </div>
+            </article>
+        @endforeach
+
+    </div>
+
+{{--   Post Your answer --}}
+    <div class="d-flex flex-column">
+        <div>
+            <form action="/answers/{{$question->id}}" method="post">
```
Comment (faizan-khalid-brainx) [inline resources/views/question.blade.php]: should use named routes

File: resources/views/questions.blade.php [hunk 14]
```diff
@@ -9,8 +9,15 @@
             </a>
         </div>
     </div>
-    <div class="question-summary d-flex p-3 ">
-        <div class="p-2">votes and answers</div>
-        <div class="p-2 bg-info flex-fill">Question Title</div>
-    </div>
+    @foreach($questions as $q)
```
Comment (faizan-khalid-brainx) [inline resources/views/questions.blade.php]: inappropriate name

File: resources/views/questions.blade.php [hunk 15]
```diff
@@ -9,8 +9,15 @@
...
+    <article class="row px-1 py">
+        <div class="col-sm-2">
+            <div>{{$q->vote}} votes</div>
+            <div>{{$q->answers->count()}} answers</div>
+        </div>
+        <div class="col-sm-10">
+            <h2> <a href="questions/{{$q->id}}"> {{$q->title}} </a></h2>
```
Comment (faizan-khalid-brainx) [inline resources/views/questions.blade.php]: use named route and pass parameters

File: resources/views/questions.blade.php [hunk 16]
```diff
@@ -9,8 +9,15 @@
...
         </div>
     </div>
-    <div class="question-summary d-flex p-3 ">
-        <div class="p-2">votes and answers</div>
-        <div class="p-2 bg-info flex-fill">Question Title</div>
-    </div>
+    @foreach($questions as $q)
+    <article class="row px-1 py">
+        <div class="col-sm-2">
+            <div>{{$q->vote}} votes</div>
```
Comment (faizan-khalid-brainx) [inline resources/views/questions.blade.php]: lazy loaded

File: resources/views/questions.blade.php [hunk 17]
```diff
@@ -9,8 +9,15 @@
...
+            <div>{{$q->answers->count()}} answers</div>
```
Comment (faizan-khalid-brainx) [inline resources/views/questions.blade.php]: lazy loading

File: routes/web.php [hunk 18]
```diff
@@ -13,10 +16,17 @@
...
 */
 
-Route::get('/', function () {
-    return view('questions');
-});
+
+Route::get('/', [QuestionController::class, 'showAll']);
 
 Route::get('/ask', function () {
     return view('ask_question');
```
Comment (faizan-khalid-brainx) [inline routes/web.php]: no routes should be used without controller

File: routes/web.php [hunk 19]
```diff
@@ -13,10 +16,17 @@
 |
 */
 
-Route::get('/', function () {
-    return view('questions');
-});
+
+Route::get('/', [QuestionController::class, 'showAll']);
```
Comment (faizan-khalid-brainx) [inline routes/web.php]: name the route

File: routes/web.php [hunk 20]
```diff
@@ -13,10 +16,17 @@
...
 
 Route::get('/ask', function () {
     return view('ask_question');
 });
+
+Route::get('/questions/{id}', function ($id) {
```
Comment (faizan-khalid-brainx) [inline routes/web.php]: no route should be without controller

File: routes/web.php [hunk 21]
```diff
@@ -13,10 +16,17 @@
...
+    $q = Question::with('answers')->findOrFail($id);
```
Comment (faizan-khalid-brainx) [inline routes/web.php]: inappropriate variable name q

File: routes/web.php [hunk 22]
```diff
@@ -13,10 +16,17 @@
...
+    return view('question', ['question' => $q]);
+});
+
+Route::post('/ask-question', [QuestionController::class, 'add']);
```
Comment (faizan-khalid-brainx) [inline routes/web.php]: routes unnamed

File: routes/web.php [hunk 23]
```diff
@@ -13,10 +16,17 @@
...
+Route::post('/answers/{id}', [AnswerController::class, 'addAns']);
```
Comment (faizan-khalid-brainx) [inline routes/web.php]: inappropriate function name

**Review by faizan-khalid-brainx — CHANGES_REQUESTED** [review]
Please fix all the commented changes

File: app/Http/Controllers/QuestionController.php [hunk 24]
```diff
@@ -7,20 +7,31 @@
...
+        $question->save();
+        return redirect()->action([QuestionController::class, 'index']);
     }
 
-    public function showAll()
+    public function create()
     {
-        $questions = Question::all();
-        return view('questions', ['questions' => $questions] );
+        return view('ask_question');
```
Comment (faizan-khalid-brainx) [inline app/Http/Controllers/QuestionController.php:27]: route name not according to conventions

File: resources/views/ask_question.blade.php [hunk 25]
```diff
@@ -2,20 +2,20 @@
...
-                <input type="text" class="form-control" id="q-title" placeholder="e.g. Is there an R function for finding a vector?" name="q-title">
+                <input type="text" class="form-control" id="questionTitle" placeholder="e.g. Is there an R function for finding a vector?" name="questionTitle">
             </div>
 
             <div class="mb-3">
-                <label class="fw-bold" for="q-body">Body:</label>
+                <label class="fw-bold" for="questionBody">Body:</label>
                 <p><small>Include all the information someone would need to answer your question</small></p>
-                <textarea class="form-control" rows="5" id="q-body" name="q-body"></textarea>
+                <textarea class="form-control" rows="5" id="questionBody" name="questionBody"></textarea>
```
Comment (faizan-khalid-brainx) [inline resources/views/ask_question.blade.php:18]: id should be in kebab cases

File: resources/views/question.blade.php [hunk 26]
```diff
@@ -1,43 +1,42 @@
...
+    <div class="d-flex flex-column py-3">
         <div>
-            <form action="/answers/{{$question->id}}" method="post">
+            <form action="{{route('answer.store', ['id'=>$question->id])}}" method="post">
                 @csrf
                 <div class="mb-3 mt-3">
-                    <h3><label for="ans-body">Your Answer</label></h3>
-                    <textarea class="form-control mt-3" rows="5" id="ans-body" name="ans-body"></textarea>
+                    <h4><label for="answerBody">Your Answer</label></h4>
+                    <textarea class="form-control mt-3" rows="5" id="answerBody" name="answerBody"></textarea>
```
Comment (faizan-khalid-brainx) [inline resources/views/question.blade.php:39]: html ids should be in kebab cases

**Review by faizan-khalid-brainx — APPROVED** [review]

File: database/seeders/DatabaseSeeder.php [hunk 27]
```diff
@@ -2,18 +2,57 @@
...
+    {
+    Question::truncate();
+    Answer::truncate();
+
+    // \App\Models\User::factory(10)->create();
+
+
+    $question = Question::create([
+    'title'=>'this is question 1 title from seeder',
+    'body' => 'this is question body from seeder',
```
Comment (faizan-khalid-brainx) [inline database/seeders/DatabaseSeeder.php]: There should be no commented code only helpful comments are allowed

**Review by faizan-khalid-brainx — APPROVED** [review]
Fix all the remaining comments and merge the PR

File: database/seeders/DatabaseSeeder.php (hunk 27 above)
Comment (muhammad-naveed-brainx) [reply database/seeders/DatabaseSeeder.php]: Ok

**Pull request merged**